app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsImageRecsFragment.kt

```diff
@@ -0,0 +1,383 @@
+        }
+
+        ImageZoomHelper.setViewZoomable(binding.imageView)
+
+        bottomSheetBehavior = BottomSheetBehavior.from(binding.bottomSheetCoordinatorLayout).apply {
+            state = BottomSheetBehavior.STATE_EXPANDED
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.CREATED) {
+                viewModel.uiState.collect {
+                    when (it) {
+                        is SuggestedEditsImageRecsFragmentViewModel.UiState.Loading -> onLoading()
+                        is SuggestedEditsImageRecsFragmentViewModel.UiState.Success -> onLoadSuccess()
+                        is SuggestedEditsImageRecsFragmentViewModel.UiState.Depleted -> onDepletedState()
+                        is SuggestedEditsImageRecsFragmentViewModel.UiState.Error -> onError(it.throwable)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        callback().updateActionButton()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (resumedMillis == 0L) {
+            resumedMillis = System.currentTimeMillis()
+        }
+    }
+
+    private fun onLoading() {
+        binding.cardItemProgressBar.isVisible = true
+        binding.cardItemErrorView.isVisible = false
+        binding.bottomSheetCoordinatorLayout.isVisible = false
+        binding.articleContentContainer.isVisible = false
+        binding.imageRecommendationsDepletedContainer.isVisible = false
+    }
+
+    private fun onError(throwable: Throwable) {
+        L.e(throwable)
+        binding.cardItemProgressBar.isVisible = false
+        binding.bottomSheetCoordinatorLayout.isVisible = false
+        binding.articleContentContainer.isVisible = false
+        binding.imageRecommendationsDepletedContainer.isVisible = false
+        binding.cardItemErrorView.isVisible = true
+        binding.cardItemErrorView.setError(throwable)
+    }
+
+    private fun onLoadSuccess() {
+        binding.cardItemProgressBar.isVisible = false
+        binding.cardItemErrorView.isVisible = false
+        binding.bottomSheetCoordinatorLayout.isVisible = true
+        binding.articleContentContainer.isVisible = true
+
+        binding.articleTitle.text = StringUtil.fromHtml(viewModel.summary.displayTitle)
+        binding.articleDescription.text = viewModel.summary.description
+        binding.articleExtract.text = StringUtil.fromHtml(viewModel.summary.extractHtml).trim()
+
+        var thumbUrl = ImageUrlUtil.getUrlForPreferredSize(viewModel.recommendation.images[0].metadata!!.thumbUrl, Constants.PREFERRED_CARD_THUMBNAIL_SIZE)
+        if (thumbUrl.startsWith("//")) {
+            thumbUrl = "https:$thumbUrl"
+        }
```

Comment:
Not sure but can we use `UriUtil.resolveProtocolRelativeUrl` here?

Reply:
sure!